### PR TITLE
Add UT for getPathFromPID

### PR DESCRIPTION
**Pull request by jichenjc**

### CODE REVIEW

File: pkg/pod_lister/pod_lister_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+func createTempFile(contents string) (filename string, reterr error) {
+	f, err := os.Create("/tmp/987654321")
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			return
```
Comment (rootfs) [inline pkg/pod_lister/pod_lister_test.go:17]: delete the temp file before exit

File: pkg/pod_lister/pod_lister_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import (
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func createTempFile(contents string) (filename string, reterr error) {
+	f, err := os.Create("/tmp/987654321")
```
Comment (rootfs) [inline pkg/pod_lister/pod_lister_test.go:11]: use `ioutil.TempFile`

File: pkg/pod_lister/pod_lister_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			contents:    ubuntu_containerd,
+			expectedRet: ubuntu_expected,
+		},
+	}
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			s := "/tmp/%d"
+			file, err := createTempFile(testcase.contents)
+			g.Expect(err).NotTo(HaveOccurred())
+			defer os.Remove(file)
```
Comment (jichenjc) [inline pkg/pod_lister/pod_lister_test.go:70]: see here @rootfs

File: pkg/pod_lister/pod_lister_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	defer func() {
+		if err := f.Close(); err != nil {
+			return
+		}
+	}()
+	_, _ = f.WriteString(contents)
+	return f.Name(), nil
+}
+
+const rhel_containerd = `
```
Comment (rootfs) [inline pkg/pod_lister/pod_lister_test.go:24]: can you add the other two cases [here](https://github.com/sustainable-computing-io/kepler/pull/152#issue-1367220950)

File: pkg/pod_lister/pod_lister_test.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		},
+	}
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			s := "/tmp/%d"
+			file, err := createTempFile(testcase.contents)
+			g.Expect(err).NotTo(HaveOccurred())
+			defer os.Remove(file)
+
+			ret, err := getPathFromPID(s, 987654321)
```
Comment (rootfs) [inline pkg/pod_lister/pod_lister_test.go:72]: can you randomize the pid? if two unit tests run in parallel, the create call will fail

**Review by rootfs — APPROVED** [review]

**Pull request merged**